src/gobupload/storage/update_table.py

```diff
@@ -0,0 +1,612 @@
+        if len(join_on) > 1:
+            join_on = [f"({self.or_join.join(join_on)})"]
+        # If both collections have states then join with corresponding geldigheid intervals
+        if self.src_has_states and self.dst_has_states:
+            join_on.extend([
+                f"(dst.{FIELD.START_VALIDITY} <= {src_ref}.{FIELD.END_VALIDITY} "
+                f"OR {src_ref}.{FIELD.END_VALIDITY} IS NULL)",
+                f"(dst.{FIELD.END_VALIDITY} >= {src_ref}.{FIELD.END_VALIDITY} "
+                f"OR dst.{FIELD.END_VALIDITY} IS NULL)"
+            ])
+        elif self.dst_has_states:
+            # If only destination has states, get the destination that is valid until forever
+            join_on.extend([
+                f"dst.{FIELD.END_VALIDITY} IS NULL"
+            ])
+
+        join_on.append(f'dst.{FIELD.DATE_DELETED} IS NULL')
+        return join_on
+
+    def _dst_table_outer_join_on(self):
+        join_on = [f"dst.{FIELD.ID} = src_dst.dst_id"]
+
+        if self.dst_has_states:
+            join_on.append(f"dst.{FIELD.SEQNR} = src_dst.dst_volgnummer")
+
+        match_seqnr = f"AND rel.dst_volgnummer = dst.{FIELD.SEQNR}" if self.dst_has_states else ""
+
+        # If a relation is found, match the dst table with the relation. Otherwise include all matches on dst
+        join_on.append(f"CASE WHEN rel.{FIELD.SOURCE_ID} IS NOT NULL "
+                       f"THEN (rel.dst_id = dst.{FIELD.ID} {match_seqnr}) "
+                       f"ELSE TRUE END")
+        return join_on
+
+    def _src_dst_join_on(self):
+        join_on = [f"src_dst.src_id = src.{FIELD.ID}"]
+
+        if self.src_has_states:
+            join_on.append(f"src_dst.src_volgnummer = src.{FIELD.SEQNR}")
+
+        join_on.append(f"src_dst.bronwaarde = {self._json_obj_ref('src')}->>'bronwaarde'")
+        return join_on
+
+    def _src_dst_select_expressions(self):
+        expressions = [f"src.{FIELD.ID} AS src_id",
+                       f"dst.{FIELD.ID} AS dst_id",
+                       f"{self._json_obj_ref('src')}->>'bronwaarde' as bronwaarde"]
+
+        if self.src_has_states:
+            expressions.append(f"src.{FIELD.SEQNR} as src_volgnummer")
+
+        if self.dst_has_states:
+            expressions.append(f"max(dst.{FIELD.SEQNR}) as dst_volgnummer")
+
+        return expressions
+
+    def _src_dst_group_by(self):
+        group_by = [f"src.{FIELD.ID}", f"dst.{FIELD.ID}", "bronwaarde"]
+
+        if self.src_has_states:
+            group_by.append(f"src.{FIELD.SEQNR}")
+        return group_by
+
+    def _where(self):
+        """WHERE clause filters all rows where src has an empty bronwaarde and there are no matching relations.
+        In that case we can safely ignore this row.
+
+        :return:
+        """
+
+        return f"WHERE NOT (" \
+            f"src.{FIELD.ID} IS NOT NULL AND {self._source_value_ref()} IS NULL AND rel.{FIELD.ID} IS NULL) " \
+            f"AND rel.{FIELD.DATE_DELETED} IS NULL"
+
```

Comment:
I propose to change to:
f"src.{FIELD.ID} IS NOT NULL AND "
f"{self._source_value_ref()} IS NULL AND "
f"rel.{FIELD.ID} IS NULL)  AND "
f"rel.{FIELD.DATE_DELETED} IS NULL"